main/signers/lattice/Lattice/index.ts

```diff
@@ -277,8 +276,8 @@ export default class Lattice extends Signer {
       const latticeTx = compatibility.compatible ? { ...rawTx } : londonToLegacy(rawTx)
 
       const signedTx = await sign(latticeTx, async tx => {
-        const unsignedTx = this.createTransaction(index, rawTx.type, latticeTx.chainId, tx)
-        const signingOptions = await this.createTransactionSigningOptions(tx, unsignedTx)
+        const unsignedTx = this.createTransaction(index, rawTx.type, latticeTx.chainId, latticeTx)
```

Comment:
why did we change from using `tx` to `latticeTx` here? the former is the one returned by the callback so I don't think we can guarantee they are the same